js/__internal/grids/grid_core/keyboard_navigation/module.ts

```diff
@@ -1594,27 +1598,47 @@ const keyboardNavigationMembers: Partial<import('./module_types').KeyboardNaviga
   _startEditCell(eventArgs, fastEditingKey) {
     this._fastEditingStarted = isDefined(fastEditingKey);
     const editResult = this._editFocusedCell();
+    const isEditResultDeferred = isDeferred(editResult);
+    const isFastEditingStarted = this._isFastEditingStarted();
 
-    if (this._isFastEditingStarted()) {
-      if (editResult === true) {
-        this._editingCellHandler(eventArgs, fastEditingKey);
-      } else if (editResult && editResult.done) {
-        const editorValue = fastEditingKey !== FAST_EDITING_DELETE_KEY ? fastEditingKey : '';
-        editResult.done(() => this._editingCellHandler(eventArgs, editorValue));
-      }
+    if (!isFastEditingStarted) {
```

Comment:
```suggestion
    if (!isFastEditingStarted || !isEditResultDeferred && !editResult) {
```

Maybe combine these conditions into one?